src/middleware/normalize.rs

```diff
@@ -78,8 +105,11 @@ where
 
         let original_path = head.uri.path();
 
-        // always add trailing slash, might be an extra one
-        let path = original_path.to_string() + "/";
+        // Either adds a string to the end (duplicates will be removed anyways) or trims all slashes from the end
+        let path = match self.trim_last_slash {
+            false => original_path.to_string() + "/",
+            true => original_path.trim_end_matches(|c| c == '/').to_string(),
```

Comment:
I believe this is equivalent.

```suggestion
            true => original_path.trim_end_matches('/').to_string(),
```